### ISSUE TITLE
Last returning different results based on range

### ISSUE THREAD

**Issue by rickspencer3** [influxdata/influxdb#20595]
This is only happening in OSS.

```
from(bucket: "idping")
|> range(start: -100y)
|> last()
|> yield(name: "-100y")

from(bucket: "idping")
|> range(start: -1h)
|> last()
|> yield(name: "-1h")
```

Results in the following:

`#group,false,false,true,true,false,false,true,true
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,long,string,string
#default,-100y,,,,,,,
,result,table,_start,_stop,_time,_value,_field,_measurement
,,0,1970-01-01T00:00:00Z,2021-01-25T17:12:14.859915Z,2021-01-24T23:58:59.280044Z,2383,n,ctr

#group,false,false,true,true,false,false,true,true
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,long,string,string
#default,-1h,,,,,,,
,result,table,_start,_stop,_time,_value,_field,_measurement
,,0,2021-01-25T16:12:14.859915Z,2021-01-25T17:12:14.859915Z,2021-01-25T17:11:08.443276Z,3219,n,ctr

```

**Comment by rickspencer3** [comment]
When I set start: -19h, it works, when I set start: -20 hours, it does not

**Comment by rickspencer3** [comment]
I tried start: -525600m and that returned incorrect results as well

**Comment by danxmoran** [comment]
Possible duplicate of influxdata/flux#3480

**Comment by danxmoran** [comment]
Fixed by #21140
